src/main/scala/org/broadinstitute/dsde/firecloud/EntityService.scala

```diff
@@ -409,6 +408,37 @@ class EntityService(rawlsDAO: RawlsDAO, importServiceDAO: ImportServiceDAO, cwds
     }
   }
 
+  def getJob(workspaceNamespace: String, workspaceName: String, jobId: String, userInfo: UserInfo): Future[ImportServiceListResponse] = {
+    // there is no way to tell just from a jobId if the job exists in cWDS or Import Service. So, we have to try both.
+
+    // if cWDS is enabled, query cWDS for job
+    val cwdsFuture: Future[ImportServiceListResponse] = if (cwdsDAO.isEnabled) {
+      rawlsDAO.getWorkspace(workspaceNamespace, workspaceName)(userInfo) map { workspace =>
+        val cwdsResponse = cwdsDAO.getJobV1(workspace.workspace.workspaceId, jobId)(userInfo)
+        logger.info(s"Found job $jobId in cWDS")
+        cwdsResponse
+      }
+    } else {
+      Future.failed(new FireCloudExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, s"Import $jobId not found")))
```

Comment:
I'm a bit confused about the exact logic going on here - it seems like we create a `FireCloudExceptionWithErrorReport` if cWDS is not enabled - does that also happen if getJobV1 throws a 404?

Reply:
I added a code comment to explain, and changed the thrown exception to be the same one that cWDS would have thrown. I don't think it makes a functional difference since the `recoverWith` (Scala version of `catch`) handles all exceptions, but I agree it's more accurate/understandable.